### ISSUE TITLE
New Resource request: Analysis Services server

### ISSUE THREAD

**Issue by kjhosein** [terraform-providers/terraform-provider-azurerm#1734]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add a new Terraform resource to create an [Azure Analysis Services](https://azure.microsoft.com/en-us/services/analysis-services/) server.

### New or Affected Resource(s)

* new: azurerm_analysis_services_server

### Potential Terraform Configuration

```hcl
resource "azurerm_analysis_services_server" "analysis_services_svr1" {
  name = "mysvr"
  resource_group_name = "rg1"
  location = "eastus"
  pricing_tier_sku = "B2"
  admin_user = "joe@example.com"
  key_expiration = "never"
  backup_blob_container = "myblobcontainer"
  tags = {
    desc = "I am not a mule"
  }
```

### References

Appears that the Azure Go SDK has support for Analysis Services:

* https://github.com/Azure/azure-sdk-for-go/tree/master/services/analysisservices/mgmt

**Comment by r0bnet** [comment]
I would suggest to add/change some things:
```hcl
resource "azurerm_analysis_services_server" "analysis_services_svr1" {
  name = "mysvr"
  resource_group_name = "rg1"
  location = "eastus"
  sku = "B2"
  admin_users = ["joe@example.com"]
  # key_expiration = "never"  did not find that? neither in arm reference nor in go sdk
  backup_blob_container = "myblobcontainer"
  gateway_resource_id = "some_id"
  enable_power_bi_service = true
  ip_v4_firewall_rules = [
    {
      name        = "MyClientIP"
      range_start = "10.0.0.0"
      range_end   = "10.0.0.255"
    }
  ]
  tags = {
    desc = "I am not a mule"
  }
}
```

**Comment by r0bnet** [comment]
PR: https://github.com/terraform-providers/terraform-provider-azurerm/pull/3721

**Comment by tombuildsstuff** [comment]
Fixed via #3721 - thanks @r0bnet :)
